File: app/src/main/java/com/focus/focusproductivitymanager/timer/Timer.kt
package com.focus.focusproductivitymanager.timer

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDate
import java.time.LocalTime

/**
 * A data class for representing timers.
 */
@Entity(tableName = "timers")
data class Timer(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val name: String = "",
    val date: LocalDate = LocalDate.now(),
    val timeStart: LocalTime = LocalTime.of(LocalTime.now().hour + 1, 0),
    val timeEnd: LocalTime = LocalTime.of(LocalTime.now().hour + 1, 30),
    val notes: String = ""
)

File: lib/dao/timers/timer.dart
import 'package:flutter/material.dart';

class Timer {
  int id;
  String title;
  String notes;
  DateTime date;
  TimeOfDay timeStart;
  TimeOfDay timeEnd;

  Timer({
    required this.id,
    required this.title,
    required this.notes,
    required this.date,
    required this.timeStart,
    required this.timeEnd,
  });
}
